quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -268,9 +278,79 @@ SolverConfigBuildItem recordAndRegisterBuildTimeBeans(CombinedIndexBuildItem com
+            throw new IllegalStateException("Multiple classes (%s) found with a @%s annotation.".formatted(
+                    convertAnnotationInstancesToString(annotationInstances), PlanningSolution.class.getSimpleName()));
+        }
+        // Multiple classes and at least one solver config does not specify the solution class
+        List<String> emptySolutionClasses = allSolverConfig.entrySet().stream()
+                .filter(e -> e.getValue().getSolutionClass() == null)
+                .map(Map.Entry::getKey)
+                .toList();
+        if (annotationInstances.size() > 1 && !emptySolutionClasses.isEmpty()) {
+            throw new IllegalStateException(
+                    """
+                            Multiple classes (%s) found with a @%s annotation and some solver configs ([%s]) does not specify it.
+                            Maybe set the XML config file to the related solver configs, or add the missing solution classes to the XML files,
+                            or remove the unnecessary solution classes from the classpath.
+                            """
+                            .formatted(convertAnnotationInstancesToString(annotationInstances),
+                                    PlanningSolution.class.getSimpleName(),
+                                    String.join(", ", emptySolutionClasses)));
+        }
+        // Unused solution classes
+        List<String> unusedSolutionClasses = annotationInstances.stream()
+                .map(planningClass -> planningClass.target().asClass().name().toString())
+                .filter(planningClassName -> allSolverConfig.values().stream()
+                        .noneMatch(c -> c.getSolutionClass().getName().equals(planningClassName)))
+                .toList();
+        if (!unusedSolutionClasses.isEmpty()) {
+            throw new IllegalStateException(
+                    "Unused classes ([%s]) found with a @%s annotation.".formatted(String.join(", ", unusedSolutionClasses),
+                            PlanningSolution.class.getSimpleName()));
+        }
+        // Validate the solution classes target types
+        List<String> invalidTargetClasses = annotationInstances.stream()
+                .filter(planningClass -> allSolverConfig.values().stream()
+                        .anyMatch(
+                                c -> c.getSolutionClass().getName().equals(planningClass.target().asClass().name().toString())))
+                .map(AnnotationInstance::target)
+                .filter(t -> t.kind() != AnnotationTarget.Kind.CLASS)
+                .map(t -> t.asClass().name().toString())
+                .toList();
+        if (!invalidTargetClasses.isEmpty()) {
+            throw new IllegalStateException("The targets classes ([%s]) with a @%s must be a class."
+                    .formatted(String.join(", ", invalidTargetClasses), PlanningSolution.class.getSimpleName()));
+        }
```

Comment:
What is a target? Nothing in the solver is called a target. The exception should not mention a "target" because the user does not know what that is.

Reply:
The validation process involves checking the Java type of the solution class, which uses the `AnnotationInstance#target()` method. The target word comes from this method and it will be removed from the error message.